token/services/selector/sherdlock/manager.go

```diff
@@ -24,29 +34,62 @@ type tokenSelectorUnlocker interface {
 
 type manager struct {
 	selectorCache utils.LazyProvider[transaction.ID, tokenSelectorUnlocker]
+	locker        Locker
+	evictionDelay time.Duration
+	cleanupPeriod time.Duration
 }
 
 type iterator[k any] interface {
 	Next() (k, error)
 	Close()
 }
 
-func NewManager(tokenDB TokenDB, lockDB LockDB, m *Metrics, precision uint64, backoff time.Duration) *manager {
-	fetcher := newMixedFetcher(tokenDB, m)
-	return &manager{
+func NewManager(
+	tokenDB TokenDB,
+	locker Locker,
+	metrics *Metrics,
+	precision uint64,
+	backoff time.Duration,
+	evictionDelay time.Duration,
+) *manager {
+	fetcher := newMixedFetcher(tokenDB, metrics)
+	m := &manager{
+		locker:        locker,
+		evictionDelay: evictionDelay,
 		selectorCache: utils.NewLazyProvider(func(txID transaction.ID) (tokenSelectorUnlocker, error) {
-			return NewSherdSelector(txID, fetcher, lockDB, precision, backoff), nil
+			return NewSherdSelector(txID, fetcher, locker, precision, backoff), nil
 		}),
+		cleanupPeriod: cleanupPeriod,
+	}
+	if cleanupTickPeriod > 0 {
+		go m.cleaner()
 	}
+	return m
 }
 
 func (m *manager) NewSelector(id transaction.ID) (token.Selector, error) {
 	return m.selectorCache.Get(id)
 }
 
 func (m *manager) Unlock(id transaction.ID) error {
+	return m.locker.UnlockByTxID(id)
+}
+
+func (m *manager) Close(id transaction.ID) error {
 	if c, ok := m.selectorCache.Delete(id); ok {
-		return c.UnlockAll()
+		return c.Close()
+	}
+	return errors.New("selector for " + id + " not found")
+}
+
+func (m *manager) cleaner() {
```

Comment:
This we could do also with a DB job (for postgres). This way we don't have to run the cleanup code for each replica.